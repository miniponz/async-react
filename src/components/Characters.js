
import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

function Characters({ characters }) {
  const charList = characters.map((character) => (
    <li key={character.name}>
      <Character name={character.name} status={character.status} species={character.species} image={character.image} />
    </li>
  ));

  return (
    <ul>
      {charList}
    </ul>
  );
}

Characters.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Characters;
