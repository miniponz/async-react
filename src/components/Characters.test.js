import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('characters component', () => {
  it('renders some characters', () => {
    const wrapper = shallow(<Characters characters={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });   
});
