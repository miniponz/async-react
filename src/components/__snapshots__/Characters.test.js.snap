// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`characters component renders some characters 1`] = `<ul />`;
